typed_sql/lib/src/typed_sql.expr_ext.dart

```diff
@@ -0,0 +1,764 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+part of 'typed_sql.dart';
+
+extension ExpressionNull on Expr<Null> {
+  /// Cast as [Expr<int?>] using `CAST(NULL AS BIGINT)`.
+  ///
+  /// This method is rarely necessar, as you can generally pass `literal(null)`
+  /// to any method expecting an `Expr<int?>`. But in some complex scenarios
+  /// this can alleviate type inference issues.
+  Expr<int?> asInt() => CastExpression._(this, ColumnType.integer);
+
+  /// Cast as [Expr<String?>] using `CAST(NULL AS TEXT)`.
+  ///
+  /// This method is rarely necessar, as you can generally pass `literal(null)`
+  /// to any method expecting an `Expr<String?>`. But in some complex scenarios
+  /// this can alleviate type inference issues.
+  Expr<String?> asString() => CastExpression._(this, ColumnType.text);
+
+  /// Cast as [Expr<double?>] using `CAST(NULL AS DOUBLE PRECISION)`.
+  ///
+  /// This method is rarely necessar, as you can generally pass `literal(null)`
+  /// to any method expecting an `Expr<double?>`. But in some complex scenarios
+  /// this can alleviate type inference issues.
+  Expr<double?> asDouble() => CastExpression._(this, ColumnType.real);
+
+  /// Cast as [Expr<bool?>] using `CAST(NULL AS BOOLEAN)`.
+  ///
+  /// This method is rarely necessar, as you can generally pass `literal(null)`
+  /// to any method expecting an `Expr<bool?>`. But in some complex scenarios
+  /// this can alleviate type inference issues.
+  Expr<bool?> asBool() => CastExpression._(this, ColumnType.boolean);
+
+  /// Cast as [Expr<DateTime?>] using `CAST(NULL AS TIMESTAMP)`.
+  ///
+  /// This method is rarely necessar, as you can generally pass `literal(null)`
+  /// to any method expecting an `Expr<DateTime?>`. But in some complex scenarios
+  /// this can alleviate type inference issues.
+  Expr<DateTime?> asDateTime() => CastExpression._(this, ColumnType.dateTime);
+
+  /// Cast as [Expr<Uint8List?>] using `CAST(NULL AS BLOB)`.
+  ///
+  /// This method is rarely necessar, as you can generally pass `literal(null)`
+  /// to any method expecting an `Expr<Uint8List?>`. But in some complex scenarios
+  /// this can alleviate type inference issues.
+  Expr<Uint8List?> asBlob() => CastExpression._(this, ColumnType.blob);
+
+  // TODO: Generate cast for CustomDataType!
+}
+
+extension ExpressionNullable<T> on Expr<T?> {
+  // TODO: Consider renaming assertNotNull -> asNotNull()
```

Comment:
```suggestion
  // TODO: Consider renaming assertNotNull -> assumeNotNull()
```

I would expect `assert` to be enforced.

Reply:
all the other casts are `asInt()` and `asString()`

Some of them like state that they are safe, other say they are not.

Maybe this is just a cast.

Reply:
but yes, 'assert' is a bad word.